src/__tests__/i18n/__fixtures__/pages/another.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { useRouter } from 'next/router';
+
+export default function Another() {
+  const { locale, locales, defaultLocale, push } = useRouter();
+  return (
+    <>
+      <h1>Another page</h1>
+      <span>{JSON.stringify({ locale, locales, defaultLocale })}</span>
+      <button
+        onClick={() => {
+          push('/', '/', { locale: 'nl-NL' });
```

Comment:
I realised by now NextJS documentation is not very detailed when it comes to things like this. You can see the signature of push here:

```js
export type PushHandler = (
  url: NextPushArgs[0],
  as: NextPushArgs[1],
  options: TransitionOptions
) => void;
```

We're definitely not using the `as` currently, and I didn't really investigate what it does.

Reply:
`as` is only used to visit a route while `history.push`ing a different path. Useful for contextual routing. I know this because I happed to write this hook :)
https://github.com/toomuchdesign/next-use-contextual-routing

Good for me, I didn't think of checking out Next types 👍